lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -356,8 +357,16 @@ def git_info_for_course(self, cdir):
 
         cmd = ['git', 'log', '-1',
                '--format=format:{ "commit": "%H", "author": "%an %ae", "date": "%ad"}', ]
+
+
+        key = 'CourseGitLog:%s' % cdir
+
         try:
-            output_json = json.loads(subprocess.check_output(cmd, cwd=gdir))
+            log_json = cache.get(key, version=version)
```

Comment:
According to doc string method ```git_info_for_course``` it  gets information like last commit etc of course.
You are taking ```version``` as a second when course update thats why ```version``` can be same for 2 commits and you can get previous version after current version. Thats why I think this code is ambitious.

also see below comment 
what do you think @pdpinch